pkgs/games/anki/sync-server.nix

```
@@ -0,0 +1,34 @@
+{ lib
+, rustPlatform
+, anki
+
+, openssl
+, pkg-config
+, protobuf
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "anki-sync-server";
+  version = anki.version;
+  src = anki.src;
+  cargoLock = anki.cargoLock;
+
+  # only build sync server
+  cargoBuildFlags = [
+    "--bin"
+    "anki-sync-server"
+  ];
+
+  nativeBuildInputs = [ protobuf pkg-config ];
+  env = {
+    PROTOC = "${protobuf}/bin/protoc";
+  };
+  buildInputs = [ openssl ];
+
+  meta = with lib; {
+    description = "Standalone official anki sync server";
+    homepage = "https://apps.ankiweb.net";
+    license = with licenses; [ agpl3Plus ];
+    maintainers = with maintainers; [ martinetd ];
+  };
```

Seems as if it should be possible to build on all platforms.

```suggestion
  meta = with lib; {
    description = "Standalone official anki sync server";
    homepage = "https://apps.ankiweb.net";
    license = with licenses; [ agpl3Plus ];
    platforms = platforms.all;
    maintainers = with maintainers; [ martinetd ];
    mainProgram = "anki-sync-server";
  };
```

---

I thought platforms.all was the default, but it's the other way around... `If no platform is specified, the package does not have prebuilt binaries.`
Thanks!

---

I'm pretty sure they are set by default in `buildRustPackage`:

https://github.com/NixOS/nixpkgs/blob/369fdb3be8dfd6ce45a61152d161b2c9c0c8a873/pkgs/build-support/rust/build-rust-package/default.nix#L159-L170

---

Builders setting platforms (and meta attributes in general) is news to me. I guess I'll have to go look through them all now :sweat_smile: